packages/core/admin/admin/src/pages/Settings/pages/ApiTokens/EditView/EditViewPage.tsx

```diff
@@ -207,48 +238,56 @@ const ApiTokenCreateView = () => {
             }),
       });
 
-      trackUsageRef.current(isCreating ? 'didCreateToken' : 'didEditToken', {
+      trackUsage(isCreating ? 'didCreateToken' : 'didEditToken', {
         type: apiToken.type,
         tokenType: API_TOKEN_TYPE,
       });
     } catch (err) {
-      const errors = formatAPIErrors(err.response.data);
-      actions.setErrors(errors);
-
-      if (err?.response?.data?.error?.message === MSG_ERROR_NAME_TAKEN) {
-        toggleNotification({
-          type: 'warning',
-          message: err.response.data.message || 'notification.error.tokennamenotunique',
-        });
-      } else {
-        toggleNotification({
-          type: 'warning',
-          message: err?.response?.data?.message || 'notification.error',
-        });
+      if (err instanceof AxiosError && err.response) {
+        const errors = formatAPIErrors(err.response.data);
+        actions.setErrors(errors);
+
+        if (err?.response?.data?.error?.message === MSG_ERROR_NAME_TAKEN) {
+          toggleNotification({
+            type: 'warning',
+            message: err.response.data.message || 'notification.error.tokennamenotunique',
+          });
+        } else {
+          toggleNotification({
+            type: 'warning',
+            message: err?.response?.data?.message || 'notification.error',
+          });
+        }
       }
+
+      // @ts-expect-error context assertation
       unlockApp();
     }
   };
 
-  const [hasChangedPermissions, setHasChangedPermissions] = useState(false);
+  const [hasChangedPermissions, setHasChangedPermissions] = React.useState(false);
 
-  const handleChangeCheckbox = ({ target: { value } }) => {
+  const handleChangeCheckbox = ({ target: { value } }: PseudoEvent) => {
```

Comment:
because all your handlers go to the Context, would it make sense to do: `handleChangeCheckbox: ContextValue['onChangeCheckbox']` for all these then we know they're in sync and avoids extra types?